src/install/repository.zig

```diff
@@ -87,4 +99,164 @@ pub const Repository = extern struct {
             }
         }
     };
+
+    fn exec(allocator: std.mem.Allocator, env: *DotEnv.Loader, cwd_dir: std.fs.Dir, argv: []const string) !string {
+        const buf_map = try env.map.cloneToEnvMap(allocator);
+        const result = try std.ChildProcess.exec(.{
```

Comment:
There's a way we can do this same thread non-blocking but it's more complicated

Reply:
I explored that a little &minus; basically `if (child_process.term) |term| ...` in some form of loop &minus; but can't convince myself the benefit was worth the trouble :sweat_smile:

Reply:
we'd use Subprocess from Bun.spawn to do that instead of going through std.ChildProcess